specification/apimanagement/developer-services-api/models/common.tsp

```diff
@@ -0,0 +1,84 @@
+import "@typespec/rest";
+import "@azure-tools/typespec-azure-core";
+
+using TypeSpec.Http;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.Core.Foundations;
+using TypeSpec.Versioning;
+
+@useDependency(Azure.Core.Versions.v1_0_Preview_2)
+namespace DeveloperServicesApi.Models;
+
+//API response for 200 with content
+alias ApiResponse<T> = T | ErrorResponse;
+
+//API response for 201 with content
+alias CreatedApiResponse<T> = Created<T> | ErrorResponse;
+
+@doc("Collection object for listing operations.")
+model Collection<T> {
+  @doc("Relative URL to next page of collection.")
+  nextLink: string;
+
+  @doc("Collection itself.")
+  value: T[];
+}
```

Comment:
You should use the models from Azure Core. Reimplementing these common types only creates opportunities for inadvertent errors, as is the case here -- `nextLink` should be optional.